### ISSUE TITLE
What does this project for?

### ISSUE THREAD

**Issue by svetlyak40wt** [michaeljforster/destore#1]
I found it searching for postmodern usage examples. And wondering that it will be, what is the purpose of the `destore`?

**Comment by michaeljforster-zz** [comment]
Thanks for the interest!  It implements Greg Young's "event store" or "event sourcing" idea: http://cqrs.nu/Faq/event-sourcing. Many other references, but that one gives a brief overview. We use it in a number of our production systems, but I haven't finished documenting and committing the higher level API. So you see only the Postmodern bindings so far.

**Comment by svetlyak40wt** [comment]
Interesting! I have a some sort of event store in my current project, but we also store a current state of each object and have no algorithm, to rebuild state from the log.

But actually, we call these "events" — "actions". User acts – we save this "action" to the log. And the action leads to one or more consequences, we call them "events". For example, operation `PATCH /user/` will be saved as an action `user_modify`, and event `user_modified`. But if field `department_id` was changed, then additional events will be triggered, like: `user_department_changed`, `department_user_remove`, `department_user_added`.

We've made this system for security reasons and to be able to notify different consumers about events they are care about.
